### ISSUE TITLE
Accessibility-> Link to expand the filter sections missing link purpose for users using screen readers

### ISSUE THREAD

**Issue by manuji** [bcgov/nr-rec-resources#657]
**Describe the bug**
On Find a site or trail page, the link given to expand filter sub sections are not descriptive enough for the users using the screen readers. This issue fails the [2.4.4 Link Purpose (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/link-purpose-in-context.html) criterion.

**To Reproduce**
1. Go to '[...](https://d1vrz2gt4nmfe5.cloudfront.net/search)'
2. Using a screen reader navigate through the Filters setion'
3. Notice how the link to "Show all xx" is read by the screen reader. 
4. It only reads "Show all xx".

**Expected behavior**
The link should be descriptive enough so that a user using only a screen reader can understand the purpose of the link. 

**Screenshots**

<img width="1467" alt="Image" src="https://github.com/user-attachments/assets/4dc38c2b-a861-4a5e-b64f-c90d05f4a5d7" />

**Additional context**
Tested on MacOs with Chrome and VoiceOver.
